crates/biome_js_analyze/src/semantic_analyzers/correctness/use_hook_at_top_level.rs

```diff
@@ -299,48 +310,65 @@ impl Rule for UseHookAtTopLevel {
         let options = ReactExtensiveDependenciesOptions::new(options.clone());
 
         let FunctionCall(call) = ctx.query();
-        let hook_name_range = call.callee().ok()?.syntax().text_trimmed_range();
-        if react_hook_configuration(call, &options.hooks_config).is_some() {
```

Comment:
I have replaced `hook_name_range` with a `get_hook_name_range` closure so that the trimmed range doesn't get calculated if there is nothing to report.

I also replaced the outer `if` with an early return to avoid another level of indentation inside most of this function.